app/controllers/pieces_controller.rb

```diff
@@ -13,20 +13,19 @@ def update
         game_id: current_piece.game_id,
         rank: current_piece.rank,
         file: current_piece.file
-      )
-      @game = current_piece.game
-      if @game.check?(current_piece)
-        flash[:alert] = "Check!"
-      end
-    else
-      return render plain: 'Not Valid', status: :unprocessable_entity
+        )
+    end
+
+    @game = current_piece.game
+    if @game.check?(current_piece)
+      flash[:alert] = "Check!"
     end
 
-    redirect_to game_path(current_piece.game.id)
```

Comment:
Removing this line of code breaks 3 of the pieces_controller_spec tests and outputs the following error:

`ActionController::UnknownFormat:
       PiecesController#update is missing a template for this request format and variant.`

According to this post https://stackoverflow.com/questions/42544805/rails-ajax-chat-app-missing-a-template-for-this-request-format-and-variant , I think this works similar to the create action in that it **renders HTML by default**

This should be resolved by either adding that line back in causing the render/redirect which we probably don't want.  Then again another feature that won't work probably is that check flash message since it can be only show on rendering the browser. I'll add another trello card to use some JS to throw a flash message that doesn't require rendering the browser.

Otherwise, in the post they did mention something about overriding the way action renders by changing the format.  Not sure how to go about this one but I'll leave that to you to figure out